### ISSUE TITLE
Hardware hacking/ circuit bending to get midi

### ISSUE THREAD

**Issue by DubbyLabby** [giveabit/Trio-Plus-Data#2]
Hi,
I just found this pedal today and I found it superb. It has obvious lacks and this project seems the right place to share my knowledge and join efforts.
It could be possible to take some detailed pictures of its internals? I think it may be possible to extract midi data (including clock) with some arduino/teensy work...

Thanks
D.

**Comment by giveabit** [comment]
Hi there,
thanks for offering your help. This is much appreciated. Unfortunately other things occupy me so much that I stopped actively developing on this project. But maybe someone else could take the lead on bit-banging with the help of a microcontroller board?!

Nevertheless - thanks a lot!!!!

**Comment by DubbyLabby** [comment]
Hi again!
I finally bought one for myself ;)
I wasn’t thinking on open it but maybe if someone see this topic and wants to take the gauntlet I could make a look into it..,

Mmmm this trigger my punk side heehehe

**Comment by Gubins53** [comment]
DubbyLabby, I have one of these. I'm still learning how to use it and I'm not a higher level programmer. I have written a lot in assembly for fast tight code over the years. Mainly for equipment real time work. Anyway, I'm in electronic engineering, I'm happy to open mine since the warranty expired while I was away. I can take pics if that is what you are after. Let me know what you want.

**Comment by DubbyLabby** [comment]
Hi Gubins53,
I finally opened it and take some pictures (but I forgot where I put these to be fair XP).
There weren't clear spots to tinker but I don't remember if I could get nice picture even on the chips. I suppose some datasheet researching can put some light on the usual questions like dedicated audio part outputs or just a synth/sampler chip with stereo pair... there's some TX/RX or SPI between these chips with some nice midi packets? etc...
I even wrote a mail to Digitech to know if they were working on mk2 with some extra features (like midi sync...). This machine is near perfect and useless at the same time... shame...

**Comment by Gubins53** [comment]
Hi DubbyLabby
XP isn't so bad. I still have a PC running it but isolated from the internet. Has its uses for old useful programs. I use it for graphics and music still.
I'll open mine up this week and takes some pics. I'll pass them on to you. I'll look up data sheets too. See what we can sort out.
Stay safe from the virus.
I agree it's a good little box but just short of some great features.

**Comment by DubbyLabby** [comment]
looool XP was a smiley about my memory haha but thanks a lot for your worries <3
I'm on mac and win10 (and my brother and some friends are Linux skilled so we can reach them if we need something on that field)
Yes please send (or upload to imgur) these pics so others can take a look. I love to tinker in this things and also I have some solutions for the midi side for those of you who have a iDevice. Not so straightforward like Trio+ but useful to get midi bass and drums...

https://www.youtube.com/watch?v=EDyraaf0yDM&t=685s

Cheers!

**Comment by Gubins53** [comment]
OK I took some pics of the main board. I assume the top board is just for the various controls. I've attached them. I've included a txt file for the list of chips. I'm pretty sure I've read the characters correctly. Correct me if I'm wrong.
I hope this helps and is a start. As for taping in to midi it seems its all firmware driven and saved to memory. I'm open to any thing you can say on this.


![Main Chips](https://user-images.githubusercontent.com/55730869/77722074-9f3ac900-703c-11ea-8bdc-3e1a40641f95.jpg)
![Main PCB](https://user-images.githubusercontent.com/55730869/77722085-a4981380-703c-11ea-8136-fb9a09cef1a2.jpg)
[Trio plus chip list.txt](https://github.com/giveabit/Trio-Plus-Data/files/4391593/Trio.plus.chip.list.txt)

**Comment by DubbyLabby** [comment]
I will look into these sure!
I remember similar conclusion about midi and so when I opened mine but I can't remember where I put the pics I took... I will research into the chips if we can get some extra data and maybe some modding to get midi or useful things.

Thanks a lot!

**Comment by Gubins53** [comment]
I’ll see if this will get to you without logging in. I’m lazy.

I looked at the data sheets for these except for the last chip, U6. It seemed to be like a 555 but not sure.

All datasheets are out there and why I figured there was no midi hardware, or so I believe.

 

A Midi file would be made and saved on the removable mem card. 

I guess the drums and bass are processed in the AD chip and fed to a DAC to send out to hear them.

A midi copy would be saved to file on the removable mem card.

So actual midi signals might not exist.

But if you can see some way from the data sheets it would be great.

I’ll go back and look at the AD chip where it all happens.

 

Fingers crossed.

cheers

**Comment by DubbyLabby** [comment]
Yes I'm looking at the datasheets and you are probably right about all the internal _cooking_ about data. Looking into the files is the best source for midi info about patters and so... but I still have an idea for those who want to take some benefit from the trio+ and need some kind of midi clock output (and maybe some other _goodies_) 
It could be possible attach an external arduino (or teensy) _sniffer_ to the ALT TIME led (or even realtime audio) and convert that info into external clock:
https://www.instructables.com/id/Sound-to-MIDI-Converter/

It also opens the possibility to add some extra sensors (LDR) to the SONG PART leds (from the side without interfering with them ) to also give some extra cue notes to external hardware even with the quantisation from the Trio/+ itself. It could implement a "latency" encoder with display to dispatch the midi data early and improve synchronisation from devices too.

I can thing in other interesting modifications for the Trio+ itself like add extra mute buttons or faders instead knobs but these are more to transform it into a tabletop player more than the simple approach of extra device add-on.

It's a shame Digitech build it so close in networking terms but I've send time ago a mail to them for an mk2 device with some of these suggestions. Let's hope they include some and even more if they release a new version someday.

**Comment by Gubins53** [comment]
I understand your ideas. But instead of light sensing, like LDRs etc, It is possible to simply connect to the switched side of the led and read that. It’s simpler, faster and accurate and not prone to ambient light conditions. A simple interface can be made if a different earth is used. The different colours could be monitored too on multi coloured LEDs.

 

Sound to midi should be easy for drums and bass, even run them separately if mixed gives any messy output from the converter. Simple enough.

You can convert the guitar parts too and run that separately with bass and drums turned down if it’s too complex for the converter all at once.

 

Good ideas.

I’ll have to look in to the Arduino. I’ve used Microchip products in the past and Intel before that.

The link you gave shows a very basic project. There must be better ones. 

The video seemed to show latency issues and occasional misses.

 

Anyway. It sounds like you have some fine ideas about this.

 

An Aussie muso, Tash Sultana, uses a BOSS RC-30. It has twin stereo tracks and more effects etc and a USB output that works.

It’s around the same price as Trio+. This girl uses two of these I believe. She does some amazing stuff and is on You tube. Worth catching just to hear her playing.

 

Cheers

Barry

**Comment by DubbyLabby** [comment]
Yupo but I was wondering something detachable to avoid solder directly for those who want to sell later the trio+  without need to undo the soldering work ;)

I know the rc10r (it’s relatively new) and I’m the moment using lumbeat apps (iBassist and drummers). These haven’t “learn” from audio like trios but can send midi (iBassist and few drummers ATM but dev has promissed share features across them) and also I suggest him to add midi/xml import from iReal Pro app to make covers a two step process :P

Back on trio around my little research I found a fb group sharing songs and also people using kaoss pad to get the audio to clock conversion (I forgot these little wonders) and obviously there are also iOS apps with similar cappabilities (but I prefer hardware for live gigs).

Since my goal was looping and song parts I have also other paths. I have GroupTheLoop app which is a little marvel and also I finally bought an rc505. Both can loop and switch parts like trio+ (among others like electroarmonix loopers) so feeding them with lumbeat parts (or samples) makes them a perfect backing track machine (since I’m so-so keyboard player I ditch the idea of realtime “teaching” the trio to just produce at home and... well stream... even before the outbreak lol)

Anyways I will love see updates on the trio (I even consider buying BiB sometimes) and even go deeper and make these _hacks_ you pointed and even more like the knob/fader conversion or cc/note from different leds but I usually open things and these get unfinished in some corner at the lab... I’m lazy and not so qualified in coding terms so that’s why external solution came to my mind and why I ended going other routes...
Oh money and free time to mess with all this lovely stuff... I hope mk2 is near and Universal Basic Income too... so I can finally open a tinkering space with some pals at my town! Hehehe

Dreaming is (still) free... <3

Enviado desde mi iPhone

**Comment by Gubins53** [comment]
I can see what you’re trying to do now in regards to staying external. In terms of re-sale a small connector could be added at the back to bring out signals from LEDs etc. and if done with care can look fine and not detract from a re-sale. I’ve made mods to other equipment like that without any problems. But what you suggest is fine too.

 

I don’t play gigs at the moment. So upgrading and making changes isn’t a priority but I do like to keep an eye on things.

 

I hope you don’t mind me asking but what country are you in?

**Comment by DubbyLabby** [comment]
Hi,
I’m (confined) at Barcelona, Spain.

Hope you and everyone reading this is fine with the coronavirus outbreak...

Blessings <3 

Enviado desde mi iPhone

**Comment by Gubins53** [comment]
Hi, I thought it was on that area from your accent (a little joke there).

I’ve been meaning to learn Spanish for years and still haven’t. I still hope to get there anyway.

 

Yes I’m keeping isolated and many are. Some people don’t seem to care but now the police here in Australia are issuing large fines to those people.

 

Take care and stay safe.
Barry

**Comment by DubbyLabby** [comment]
U2 mate <3

**Comment by merelyhumanmusic** [comment]
Hey, friends.  I am looking to do almost exactly this with my Trio+: cutting/snooping the alt time LED current signal, analyzing with an arduino, then outputting a 2 Pulse-Per-Quarter-Note signal to syncronize a set of Pocket Operators to the Trio+ (maybe expanded to midi clock too in the future).  Did either of you make any progress?  Can I help?

**Comment by Gubins53** [comment]
Hi, No I didn’t explore this any further. In fact I shelved my unit and only just started using it again.

Good luck with your plans. You sound like you know what you’re doing, just watch for static damage.

 

Cheers

Barry

**Comment by DubbyLabby** [comment]
Hi guys! (Hope you are well <3)
I gone finally the lumbeat apps route since it brings me more control.
I sent a mail to digitech back in the day with some suggestions for an mk2 version but also check Beatbuddy pedal if you need standalone hardware solution ready. 
I dream time to time about improving a trio+ for further jamming but since I want to do more reggae/dub in the future the lumbeat apps was just the most straight forward solution for my needs.

If you want to share your journey and I can help somehow, I will try to contribute.


Cheers.

**Comment by DubbyLabby** [comment]
Hi again!

Time goes by... but madness stays! ;)

First of all, I hope all of you are health and happy. I still struggle with family matters (call me Urkel) but I recently get another Trio+ and some new ideas (about setup and about modding)

So after join a FB group and see some users perform some of the mods we were discussing time ago I decided to open mine and take more photos. I took detailed ones about the "control board" since someone suggested hijack the ribbon cable signal to perform the mods. I thought it was a clever idea but my gut said "check that pcb... it could not be so easy" and it was right.

Long story short... that ribbon is a serial bus between the Main Board and shift register based matrix (74HC595 x8 and 74HC168D x2) so it will require first of all reverse engineer the schematic to later listen that bus and try to trick the Main Board to make some magic as midi input/output, etc...

So enough for today. Here the pics!

![Image](https://github.com/user-attachments/assets/2aac9237-5554-4311-a4c1-f9130e01b98a)

![Image](https://github.com/user-attachments/assets/1047374b-4498-4373-9e57-189986f1ef17)

![Image](https://github.com/user-attachments/assets/ced0ba7e-a402-4d62-9e0d-f6b4a3f87fcd)

**Comment by DubbyLabby** [comment]
![Image](https://github.com/user-attachments/assets/b57ef28b-b17a-4534-803d-77d38ae5ac35)

![Image](https://github.com/user-attachments/assets/593dd638-46fe-4000-8f0e-3aa9673d8a57)
